pkg/watcherapplier/statefulset.go

```diff
@@ -0,0 +1,137 @@
+package watcherapplier
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	watcherv1beta1 "github.com/openstack-k8s-operators/watcher-operator/api/v1beta1"
+	watcher "github.com/openstack-k8s-operators/watcher-operator/pkg/watcher"
+)
+
+const (
+	// ServiceCommand -
+	ServiceCommand = "/usr/local/bin/kolla_start"
+
+	// ComponentName -
+	ComponentName = watcher.ServiceName + "-applier"
+)
+
+// StatefulSet - returns the StatefulSet definition for the watcher-applier service
+func StatefulSet(
+	instance *watcherv1beta1.WatcherApplier,
+	configHash string,
+	labels map[string]string,
+	annotations map[string]string,
+) *appsv1.StatefulSet {
+	runAsUser := int64(0)
```

Comment:
this is wrong
```
runAsUser := int64(0)
```
should be
```
runAsUser := int64(42451)
```
https://github.com/openstack-k8s-operators/tcib/blob/main/container-images/kolla/base/uid_gid_manage.sh#L74

you can see we do  more or less the same in placement except we pull it out into a constant.

https://github.com/openstack-k8s-operators/placement-operator/blob/main/pkg/placement/const.go#L36
https://github.com/openstack-k8s-operators/placement-operator/blob/5c39b092c37feddb1d94539b759ee8cebab371e2/pkg/placement/deployment.go#L137